pages/api/hashicorp.ts

```diff
@@ -15,9 +17,50 @@ export default function handler(
 	res: NextApiResponse<ResponseData>
 ) {
 	const { query } = req
-	const searchParam = query.search || ''
+	const searchParam = (query.search as string) || ''
+	const hideNoPicture = (query.hideNoPicture as string) || ''
 
-	// Sr. candidate TODO: Perform DB query and return the result
+	try {
+		const db = new Database('hashicorp.sqlite')
```

Comment:
We are opening up a new DB connection on every request here. What would be the downsides of that and what are some other approaches?

Reply:
The downsides are that it can cause performance and scaling issues. I switched this to a singleton pattern so that all requests share the same connection